lib/marisa/exception.cc

```diff
@@ -0,0 +1,21 @@
+#include "marisa/exception.h"
+
+namespace marisa {
+
+/**
+ * This is defined here for the following reasons.
+ *
+ * 1. Virtual destructors are best not defined in a header to be included in many compilation units.
+ * Defining it here in a single compilation unit prevents multiple vtables from being generated and causing
+ * a compiler to be confused as to which is the true vtable, like with full link time optimization.
+ * By reducing the number of vtable instances, you also reduce the size of your library or application.
```

Comment:
Did you measure a size difference?

Reply:
I don't have the specific number any more, but it was non-zero.

Reply:
Interesting. I'll have to try.

Reply:
I just tried on OS X. `libmarisa.a` is about 10k smaller when `~Exception` is defined in `exception.cc`.  The size difference goes away when it's linked, and the final static `tools/` binaries are a few 10s of bytes smaller. `libmarisa.dylib` is also the same size.

I guess some sufficiently bad linker must exist somewhere though.